mlrun/k8s_utils.py

```diff
@@ -475,6 +475,37 @@ def get_project_secret_data(self, project, secret_keys=None, namespace=""):
 
         return results
 
+    def list_services(self, namespace=None, states=None):
+        try:
+            services = self.v1api.list_namespaced_service(
+                self.resolve_namespace(namespace)
+            )
+        except ApiException as exc:
+            logger.error(f"failed to list services: {exc}")
+            raise exc
+
+        return [
+            service
+            for service in services.items
+            if not states or service.status.phase in states
+        ]
+
+    def is_service_exist(self, name, namespace=None, is_release_name: bool = True):
+        items = self.list_services(namespace=namespace)
+        for item in items:
+            if (
+                is_release_name
+                and item.metadata.annotations is not None
+                and "meta.helm.sh/release-name" in item.metadata.annotations
+            ):
+                if name == item.metadata.annotations["meta.helm.sh/release-name"]:
```

Comment:
you can simply add it the the list of 'and's in the first if. And is checked from left to right and the first to return False will stop the check so if the `"meta.helm.sh/release-name"` key is not in the dictionary the `name == item.metadata.annotations["meta.helm.sh/release-name"]` won't be checked.